### PR TITLE
feat: implement greptime cli export

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

implement `greptime cli export` command line for exporting all `SHOW CREATE TABLE` under one catalog。

E.g.:
```shell
greptime cli export --addr "127.0.0.1:4001" --catalog greptime --schema public --output-dir /tmp/greptimedb-export --target create-table
```

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Comment by sunng87** [comment]
How about changing `--schema` and `--catalog` to `--database` for consistency with mysql? So we don't need to explain schema/catalog concept in our docs.

File: src/cmd/src/cli/export.rs [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+    /// Iterate over all db names.
+    ///
+    /// Newbie: `db_name` is catalog + schema.
+    async fn iter_db_names(&self) -> Result<Vec<(String, String)>> {
+        if !self.schema.is_empty() {
+            Ok(vec![(self.catalog.clone(), self.schema.clone())])
+        } else {
+            Ok(vec![])
+        }
+    }
```
Comment (shuiyisong) [inline src/cmd/src/cli/export.rs:118]: if `self.schema` is empty, then it seems to skip working

**Comment by waynexia** [comment]
> How about changing `--schema` and `--catalog` to `--database` for consistency with mysql? So we don't need to explain schema/catalog concept in our docs.

`--schema` and `--catalog` can be assigned separately, e.g., only export tables under `catalog = greptime`. Does `--database` has the same semantic?

**Comment by sunng87** [comment]
In that case we will need to support wildcard with database, like `--database=*`, `--database=greptime-*` or `--all`. Note that we only have one catalog in self-hosted version so in most case it's like `--all`.

I think it's OK to have catalog/schema as an internal tool. But if we expose it as a user-facing tool we will need to explain those concepts.

**Comment by waynexia** [comment]
@sunng87 @shuiyisong PTAL

**Review by sunng87 — APPROVED** [review]

**Comment by sunng87** [comment]
LGTM.

But on top of this we may need authentication or tls setup to secure this tool.

**Comment by killme2008** [comment]
@shuiyisong PTAL

**Comment by killme2008** [comment]
@shuiyisong PTAL

**Review by shuiyisong — APPROVED** [review]
Export create table SQL works on v0.3 clusters

**Pull request merged**